LandSandBoat/server
# [cpp/lua] Bp recast

**_I affirm:_**

## What does this pull request do?

4 things
- Deactivate was right above the bp delay code and I noticed it has leftover code from before the lua file reset the recast.
  - The code is superfluous, so I removed it
- the MP cost of Cait Sith abilities at some point was broken due to the lack of a call to `onUseBloodpact`, this is fixed (and all BP tested)
- The recast IDs for various groups of JA were never defined as an enum
- The meat of this PR:
  - Blood Pact JA recast is only set on the successful completion of a petskill (similar to the MP consumption)
  - However (here's the tricky part) it is confirmed that bp delay gear is considered on use of the player JA, not when the pet performs the action
  - I've made this functionality work properly for all smn pets that use `onUseBloodpact` in lua, and left it the same for the fallback (which I believe is only Diabolos at this point)

## Steps to test these changes

Activate an automaton, make it full hp and deactivate to see activate timer reset still

Summon various avatars, give yourself `AVATAR_PERPETUATION` to negate the perp cost, and use various blood pacts to see proper mp consumption, but also no recast packet until the skill goes off (for all but diabolos)

set `BP_DELAY` and `BP_DELAY_II` mods to reduce bp recast:
![image](https://github.com/LandSandBoat/server/assets/131182600/64be539e-8298-4e6e-aafc-8b79d7a3cb87)

https://imgur.com/qWpsnqg
(Note the mp consumption and bp ward recast happening when the skill goes off, with 45s due to having -15 bp_delay mod)

use diabolos skills and see he functions the same as before, consuming mp and recast on player JA use
